File: 0210_Problem_1.fs
// code_report Solution
// Problem Link (Contest):  https://leetcode.com/contest/weekly-contest-210/problems/maximum-nesting-depth-of-the-parentheses/
// Problem Link (Practice): https://leetcode.com/problems/maximum-nesting-depth-of-the-parentheses/

// Solution 1

let maxDepth = Seq.map(function | '(' ->  1
                                | ')' -> -1     
                                | _   ->  0)
            >> Seq.scan(+) 0
            >> Seq.max

// Solution 2

let maxDepth : string -> int = 
    Seq.map(function '(' ->  1 | ')' -> -1 | _  ->  0)
    >> Seq.scan(+) 0
    >> Seq.max

File: 0210_Problem_1.kt
// code_report Solution
// Problem Link (Contest):  https://leetcode.com/contest/weekly-contest-210/problems/maximum-nesting-depth-of-the-parentheses/
// Problem Link (Practice): https://leetcode.com/problems/maximum-nesting-depth-of-the-parentheses/

fun maxDepth(s: String)
    = s.filter{ it in "()" }
       .map{ if (it == '(') 1 else -1 }
       .scan(0){ acc, v -> acc + v }
       .max()
